store-api-server/src/payment/service/payment.service.ts

```diff
@@ -403,6 +419,162 @@ WHERE nft_order.id = $1
     };
   }
 
+  async #createSimplexPaymentIntent(
+      userAddress: string,
+      fiatCurrency: string,
+      currencyUnitAmount: number,
+      usr: UserEntity
+  ): Promise<PaymentIntentInternal> {
+    if (
+        typeof SIMPLEX_API_URL === "undefined" ||
+        typeof SIMPLEX_API_KEY === "undefined" ||
+        typeof SIMPLEX_WALLET_ID === "undefined"
+    ) {
+      throw new HttpException(
+          "Simplex payment provider not supported by this API instance",
+          HttpStatus.NOT_IMPLEMENTED
+      );
+    }
+    if (!SIMPLEX_ALLOWED_FIAT.includes(fiatCurrency)) {
+      throw new HttpException(
+          `requested fiat (${fiatCurrency}) is not supported by Wert`,
+          HttpStatus.BAD_REQUEST
+      );
+    }
+    const decimals = SUPPORTED_CURRENCIES[fiatCurrency];
+    const amount = (Number(currencyUnitAmount) * Math.pow(10, -decimals)).toFixed(
+        decimals
+    )
+    let quoteId;
+    try {
+      let quoteResponse = await axios.post(
+          SIMPLEX_API_URL + "/wallet/merchant/v2/quote",
+          {
+            end_user_id: "" + usr.id,
+            digital_currency: "USD-DEPOSIT",
+            fiat_currency: "USD",
+            requested_currency: "USD",
+            requested_amount: Number(amount),
+            wallet_id: SIMPLEX_WALLET_ID,
+            client_ip: "1.2.3.4", // TODO ?
+            payment_methods: ["credit_card"]
+          },
+          {
+            headers: {
+              "Authorization": `ApiKey ${SIMPLEX_API_KEY}`
+            }
+          }
+      );
+      quoteId = quoteResponse.data?.quote_id;
+    } catch (error) {
+        let errorMessage;
+        if (error instanceof Error) {
+            if (axios.isAxiosError(error) && error.response) {
+                errorMessage = error.response?.data?.error || error.response?.data;
+                let errors = error.response?.data?.errors;
+                if (errors && typeof errors == "object") {
+                    errorMessage = error.response?.data?.error + "---DETAILS:---" + JSON.stringify(errors);
+                }
+                console.log('get quote ERROR' + errorMessage);
+                throw new HttpException(
+                    `there is problem simplex api get quote please contact your backend services detail: (${errorMessage})`,
+                    HttpStatus.BAD_REQUEST
+                );
+            }
+        } else {
+            console.log("Unexpected error simplex api get quote");
+        }
+        throw new HttpException(
+            `there is problem simplex api get quote please contact your backend services`,
+            HttpStatus.BAD_REQUEST
+        );
+    }
+
+    const paymentId = uuidv4();
+    const orderId = uuidv4();
+
+    try {
+      var paymentResponse = await axios.post(
+          SIMPLEX_API_URL + "/wallet/merchant/v2/payments/partner/data",
```

Comment:
Actually. I got you However, Simplex works like Stripe. And it does not support Tezos. We have to put only fiat currency  :/ I wish it changes.

Reply:
That's fine that it is more like Stripe. Just my confusion, somewhere in the past I picked up the wrong assumption that Nuvei/Simplex provides a similar service as Wert does.

Anyways, then what seems to be missing is a way to get payment status updates from Simplex? For example, with stripe there is a webhook that Stripe calls everytime there is some status update of an existing payment (for example to notify us it has changed status to `processing`, or to `failed`, or to `succeeded`, etc.). Either that same approach is needed for Simplex or we should be polling them periodically ourselves. Whichever way they provide for this, basically.

Reply:
Actually, payment request provides only get paymentid for starting payment operations from webui. Then user paid with credit card. IFrame handles it. Then FE said us the operation will end succeeded or failed. https://integrations.simplex.com/wallet-api-integration please check the payment request section.
they say:
`The partner’s client calls Partner’s server API, which then: a. The partner’s backend calls Simplex’s Payment Request endpoint b. The partner’s HTTP form sends an HTTPS POST, submitted to Simplex and end-user sees payment details form, co-branded Simplex & partner.`

Reply:
At no point in this summary does the backend get informed that the payment succeeded though. Which is necessary, it's the backend that ultimately truly finalizes the payment process by giving the purchaser their NFT(s).

Just to be clear, this information cannot come from the frontend (that the payment succeeded), because the frontend is not a trusted entity. If we add such a backend API endpoint to inform that payment succeeded, someone can fool us by just manually sending this confirmation without having actually paid anything.

Therefore, payment status updates must come from Simplex themselves. Either through a webhook (we do it like this with stripe, here: https://github.com/umitduran/motion_kanvas/blob/feat/simplexapiintegration/store-api-server/src/payment/controller/payment.controller.ts#L40) or through some kind of polling mechanism to Simplex (eg request every, say, 30 seconds whether any of the payments have an updated status).

Reply:
Yeah,you are right.  I will research after payment succeeded or failed from client. Good points, thanks!